pkgs/dkg/drand.go

```diff
@@ -431,34 +799,117 @@ func (o *LocalOwner) Process(from uint64, st *wire.SignedTransport) error {
 
 		o.Exchanges[from] = exchMsg
 
-		if len(o.Exchanges) == len(o.data.init.Operators) {
+		// check if have all participating operators pub keys, then start dkg protocol
+		if o.checkOperators() {
 			if err := o.StartDKG(); err != nil {
 				return err
 			}
 		}
+	case wire.ReshareExchangeMessageType:
+		exchMsg := &wire.Exchange{}
+		if err := exchMsg.UnmarshalSSZ(t.Data); err != nil {
+			return err
+		}
+		if _, ok := o.Exchanges[from]; ok {
+			return ErrAlreadyExists
+		}
+
+		o.Exchanges[from] = exchMsg
+		allOps := append(o.Data.Reshare.OldOperators, o.Data.Reshare.NewOperators...)
+		for _, op := range allOps {
+			if o.Exchanges[op.ID] == nil {
+				return nil
+			}
+		}
```

Comment:
better approach like in normal dkg is just to check whether the len is equal.
`len(o.Exchanges) == allOps`, more efficient even though the map is small.